src/extension/inspire/src/main/java/org/geoserver/inspire/LanguagesDispatcherCallback.java

```diff
@@ -29,21 +27,8 @@ public Request init(Request request) {
             try {
                 Properties mappings = InspireDirectoryManager.get().getLanguagesMappings();
                 String isoLang = mappings.getProperty(value);
-                String supportedLanguages =
-                        mappings.keySet().stream()
-                                .map(s -> s.toString())
-                                .collect(Collectors.joining(","));
-                if (isoLang == null) {
-                    value = value == null || value.equals("") ? "empty" : value;
-                    throw new ServiceException(
-                            "A Language parameter was provided in the request but it cannot be resolved to an ISO lang code."
-                                    + " Parameter value is "
-                                    + value
-                                    + " while supported languages are "
-                                    + supportedLanguages);
-                }
                 rawKvp.put(ACCEPT_LANGUAGES_PARAM, isoLang);
-                rawKvp.put(LANGUAGE_PARAM, isoLang);
+                rawKvp.put(LANGUAGE_PARAM, isoLang != null ? isoLang : "");
```

Comment:
Is this necessary? Only seems to make the code a bit more complicated in ServicesUtils below (checking both for null and empty)

Reply:
Actually this is needed because of this change:

https://github.com/geoserver/geoserver/pull/7752/files#diff-7c86ea06a8e3960b61488a34357936b909bb33a4702523cd7e8ef10c7a7ad7b8R257-R260